### PR TITLE
P2P reception with priority

### PR DESCRIPTION
**Pull request by shargon**

### CODE REVIEW

File: neo/IO/Helper.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ internal static int GetVarSize(this string value)
         public static string ReadFixedString(this BinaryReader reader, int length)
         {
             byte[] data = reader.ReadBytes(length);
-            return Encoding.UTF8.GetString(data.TakeWhile(p => p != 0).ToArray());
+            return Encoding.UTF8.GetString(data).TrimEnd('\0');
```
Comment (erikzhang) [inline neo/IO/Helper.cs]: Think of this: "abc\0def\0".

Comment (shargon) [reply neo/IO/Helper.cs]: Maybe is better change to TrimStart?

**Comment by shargon** [comment]
Note that `inv` was changed to `invpool` when is a response of `mempool` for allow to prioritize this request

**Comment by shargon** [comment]
We need to prioritize the messages received because otherwise it is possible for consensus messages to arrive later

**Comment by erikzhang** [comment]
Many commands can be processed at approximately the same time as they are queued. So maybe we can extract those time-consuming operations into the queue or become an asynchronous operation.

**Pull request closed without merging**